sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceRDD.scala

```diff
@@ -86,6 +92,9 @@ private class PartitionIterator[T](reader: PartitionReader[T]) extends Iterator[
     if (!hasNext) {
       throw QueryExecutionErrors.endOfStreamError()
     }
+    reader.currentMetricsValues.foreach { metric =>
+      customMetrics(metric.name()) += metric.value()
```

Comment:
shall we call `+=` or `=`? `reader` keeps alive and `reader.currentMetricsValues` will keep increasing IIUC.